### PR TITLE
Fix key-name loading - [MOD-5252]

### PR DESCRIPTION
**Pull request by GuyAv46**
Small fix for loading a key name. We previously only looked at the name of the RLookup key, and therefore if one requests to load the key name as some costume name, we wouldn't understand and just ignore the load

### CODE REVIEW

File: tests/pytests/test_issues.py [hunk 1]
```diff
@@ -775,3 +775,24 @@ def test_mod5062(env):
   # verify using counter instead of sorter, even with explicit sort
   aggregate_profile = env.cmd('FT.PROFILE', 'idx', 'AGGREGATE', 'QUERY', 'hello', 'SORTBY', '1', '@t')
   env.assertEquals('Counter', aggregate_profile[1][4][2][1])
+
+def test_mod5252(env):
+  # Create an index and add a document
+  env.expect('FT.CREATE', 'idx', 'SCHEMA', 't', 'TEXT', 'n', 'NUMERIC').equal('OK')
```
Comment (oshadmi) [inline tests/pytests/test_issues.py:781]: BTW, we allow to override `__key` with an alias such as 
`FT.CREATE idx2 SCHEMA t TEXT n NUMERIC AS __key`
And then it no longer returns the key, and as requested, returns the attribute - seems OK to me.

**Review by oshadmi — APPROVED** [review]
👍🏼

File: tests/pytests/test_issues.py (hunk 1 above)
Comment (GuyAv46) [reply tests/pytests/test_issues.py:781]: I wondered if we should prevent this, or allow setting another key name, but I guess we can discuss this in the future

File: src/rlookup.c [hunk 2]
```diff
@@ -581,7 +581,7 @@ static int getKeyCommonHash(const RLookupKey *kk, RLookupRow *dst, RLookupLoadOp
     // as it will hold the only reference to it after the next line.
     rsv = hvalToValue(val, kk->fieldtype);
     RedisModule_FreeString(RSDummyContext, val);
-  } else if (!strncmp(kk->name, UNDERSCORE_KEY, strlen(UNDERSCORE_KEY))) {
+  } else if (!strncmp(kk->path, UNDERSCORE_KEY, strlen(UNDERSCORE_KEY))) {
```
Comment (rafie) [inline src/rlookup.c:584]: should we make `strlen(UNDERSCORE_KEY)` a static const or just rely on intelligence of optimizer?

**Review by rafie — APPROVED** [review]

File: src/rlookup.c (hunk 2 above)
Comment (GuyAv46) [reply src/rlookup.c:584]: I want to believe that `-O3` and even `-O2` are doing so

**Pull request merged**